src/sentry/sentry_metrics/indexer/base.py

```diff
@@ -189,6 +252,99 @@ def __getitem__(self, org_id: int) -> Mapping[str, Optional[int]]:
         return self.results[org_id]
 
 
+class UseCaseResults:
+    def __init__(self) -> None:
+        self.results: MutableMapping[str, KeyResults] = defaultdict(lambda: KeyResults())
+
+    def __eq__(self, __value: object) -> bool:
+        return isinstance(__value, self.__class__) and self.results == __value.results
+
+    def add_use_case_result(
+        self,
+        use_case_result: UseCaseResult,
+        fetch_type: Optional[FetchType] = None,
+        fetch_type_ext: Optional[FetchTypeExt] = None,
+    ) -> None:
+        self.results[use_case_result.use_case_id].add_key_result(
+            KeyResult(use_case_result.org_id, use_case_result.string, use_case_result.id),
+            fetch_type,
+            fetch_type_ext,
+        )
+
+    def add_use_case_results(
+        self,
+        use_case_results: Sequence[UseCaseResult],
+        fetch_type: Optional[FetchType] = None,
+        fetch_type_ext: Optional[FetchTypeExt] = None,
+    ) -> None:
+        for use_case, grouped_use_case_results in groupby(
+            use_case_results, lambda use_case_result: use_case_result.use_case_id
+        ):
+            self.results[use_case].add_key_results(
+                [
+                    KeyResult(use_case_result.org_id, use_case_result.string, use_case_result.id)
+                    for use_case_result in grouped_use_case_results
+                ],
+                fetch_type,
+                fetch_type_ext,
+            )
+
+    def get_mapped_results(self) -> Mapping[str, Mapping[int, Mapping[str, Optional[int]]]]:
+        return {
+            use_case_id: mapped_result
+            for use_case_id, key_results in self.results.items()
+            if (mapped_result := key_results.get_mapped_results())
+        }
+
+    def get_unmapped_use_cases(self, use_cases: UseCaseCollection) -> UseCaseCollection:
```

Comment:
Same thing here and in similar spots, maybe we can rename it to `get_unmapped_use_case_keys`

Reply:
Done